libavfilter/dnn/dnn_backend_openvino.c

```diff
@@ -173,18 +185,43 @@ DNNModel *ff_dnn_load_model_ov(const char *model_filename, const char *options)
     if (!ov_model)
         goto err;
     ov_model->ctx.class = &dnn_openvino_class;
+    ctx = &ov_model->ctx;
 
     status = ie_core_create("", &ov_model->core);
     if (status != OK)
         goto err;
 
+    //parse options
+    av_opt_set_defaults(ctx);
+    if (av_opt_set_from_string(ctx, options, NULL, "=", "&") < 0) {
+        av_log(ctx, AV_LOG_ERROR, "Failed to parse options \"%s\"\n", options);
+        goto err;
+    }
+
     status = ie_core_read_network(ov_model->core, model_filename, NULL, &ov_model->network);
     if (status != OK)
         goto err;
 
-    status = ie_core_load_network(ov_model->core, ov_model->network, "CPU", &config, &ov_model->exe_network);
-    if (status != OK)
+    status = ie_core_load_network(ov_model->core, ov_model->network, ov_model->ctx.options.device_type, &config, &ov_model->exe_network);
+    if (status != OK) {
+        a_dev = av_malloc(sizeof(ie_available_devices_t));
+        if (!a_dev){
+            av_log(ctx, AV_LOG_ERROR, "Failed to allocate memory for device info\n");
+            goto err;
+        }
+        status = ie_core_get_available_devices(ov_model->core, a_dev);
+        if (status != OK) {
+            av_log(ctx, AV_LOG_ERROR, "Failed to get available devices\n");
+            goto err;
+        }
+        for (int i = 0; i < a_dev->num_devices; i++)
+            a_dev_names = a_dev_names ? av_asprintf("%s %s", a_dev_names, a_dev->devices[i]) :
```

Comment:
a_dev_names is always NULL??

Reply:
No, if a_dev_names null, will it with a_dev->devices[0]

Reply:
No, if a_dev_names null, will fill it with a_dev->devices[0]

Reply:
i see.